Source/Methods/Envelope Tracking/EnvelopeTracking.cs

```diff
@@ -0,0 +1,309 @@
+    {
+
+        #region Type Definitions
+        public struct EnvelopeGeneratorConfiguration
+        {
+            public RfsgTerminalConfiguration TerminalConfiguration;
+
+            public static EnvelopeGeneratorConfiguration GetDefault()
+            {
+                return new EnvelopeGeneratorConfiguration()
+                {
+                    TerminalConfiguration = RfsgTerminalConfiguration.Differential,
+                };
+            }
+        }
+
+        public struct TrackerConfiguration
+        {
+            public double InputImpedance_Ohms;
+            public double CommonModeOffset_V;
+            public double Gain_VperV;
+            public double OutputOffset_V;
+
+            public static TrackerConfiguration GetDefault()
+            {
+                return new TrackerConfiguration()
+                {
+                    InputImpedance_Ohms = 1e6,
+                    CommonModeOffset_V = 1.0,
+                    Gain_VperV = 2.5,
+                    OutputOffset_V = 2.55
+                };
+            }
+        }
+
+        public enum DetroughType { 
+            Exponential, 
+            Cosine, 
+            Power
+        };
+
+        public struct DetroughConfiguration
+        {
+            public DetroughType Type;
+            public double MinimumVoltage;
+            public double MaximumVoltage;
+            public double Exponent;
+
+            public static DetroughConfiguration GetDefault()
+            {
+                return new DetroughConfiguration()
+                {
+                    Type = DetroughType.Exponential,
+                    MinimumVoltage = 1.5,
+                    MaximumVoltage = 3.5,
+                    Exponent = 1.2
+                };
+            }
+        }
+
+        public struct LookUpTable
+        {
+            public float[] DutInputPower_dBm;
+            public float[] SupplyVoltage_V;
+        }
+
+        public struct SynchronizationConfiguration
+        {
+            public double RFDelayRange_s;
+            public double RFDelay_s;
+
+            public static SynchronizationConfiguration GetDefault()
+            {
+                return new SynchronizationConfiguration()
+                {
+                    RFDelayRange_s = 2e-6, // +/- 1us
+                    RFDelay_s = 0.0
+                };
+            }
+        }
+        #endregion
+
+        #region Envelope Creation
+        public static Waveform CreateEnvelopeWaveform(Waveform referenceWaveform, DetroughConfiguration detroughConfig)
+        {
+            Waveform envelopeWaveform = new Waveform()
+            {
+                Name = referenceWaveform.Name + "Envelope",
+                Data = referenceWaveform.Data.Clone(),
+                SignalBandwidth_Hz = 0.8 * referenceWaveform.SampleRate,
+                PAPR_dB = double.NaN, // unnecessary to calculate as it won't be used
+                BurstLength_s = double.NaN, // not applicable in ET

```

Comment:
Why bother if they won't be used?

Reply:
if debugging, seeing a papr value in there would be misleading. we won't use it and setting it to 0 technically isn't true either.
I could see how we could keep the burst length the same though, so i will change it to carry it over from the reference waveform.

Reply:
bb4f330ef01c30bb5cc7d8684715c8ddf64df827